### PR TITLE
[REF] survey.stage (dynamic) replaced by static state on survey.survey

### PR DESCRIPTION
**Pull request by Feyensv**
State customization of surveys through survey.stage removed.
The users won't be able to create custom survey stages anymore.
Only 3 static states are left on survey.survey (draft, open, closed)

Form : use buttons to navigate between states instead of the clickable statusbar
Kanban : don't allow survey drag&drop anymore (long-term deletion of the kanban view?)

Task ID : 1949110

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/survey/models/survey_survey.py [hunk 1]
```diff
@@ -21,12 +21,13 @@ class Survey(models.Model):
     _rec_name = 'title'
     _inherit = ['mail.thread', 'mail.activity.mixin']
 
-    def _get_default_stage_id(self):
-        return self.env['survey.stage'].search([], limit=1).id
-
     def _get_default_access_token(self):
         return str(uuid.uuid4())
 
+    def _states(self, values, domain, order):
```
Comment (dbeguin) [inline addons/survey/models/survey_survey.py]: Name this _read_group_states.

Comment (Feyensv) [reply addons/survey/models/survey_survey.py]: Corrected, thanks :)

**Comment by Yenthe666** [comment]
Wait, why would you ever do that? This is literally removing functionality and I don't see why you'd want to remove this?

**Comment by Feyensv** [comment]
@Yenthe666 : the debate between static states or dynamic stages isn't clearly decided in Odoo :D. 
The primary spec was to update the states/stages s.t. the buttons of the survey aren't always displayed, because they led to unexpected/forbidden behavior in the current state of the module.
I proposed two solutions for this : 
1) a state (draft, open, closed) on the stages to know which actions are available for each stage (but keeping the ability for the customer to customize the survey flow with different stages) (PR #32237)
2) a fixed state (this PR)

The internal debate led to the second choice.
Ping @sbuhl

**Comment by Yenthe666** [comment]
@Feyensv thanks for the feedback and the insight. I can understand that point 2 is possible easier to change and handle but from the user experience this is a serious step back. I know plenty of people who use the survey module with dynamic states and they would loose this functionality in the master version. It is a pity to remove a functionality, no?

**Comment by Feyensv** [comment]
My preference goes to the flexibility and keeping the stages, but the final decision goes to @sbuhl :D

**Comment by Yenthe666** [comment]
@Feyensv I understand, there is always somebody above you hmm ;)
@sbuhl could you be so kind to let me/us know what is the reasoning for going with fixed stages before the flexibility of dynamic stages?

File: addons/hr_recruitment_survey/data/survey_demo.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <odoo><data noupdate="1">
     <record id="survey_recruitment_form" model="survey.survey">
         <field name="title">Recruitment Form</field>
-        <field name="stage_id" ref="survey.stage_in_progress"/>
+        <field name="state" eval="'open'" />
```
Comment (tde-banana-odoo) [inline addons/hr_recruitment_survey/data/survey_demo.xml]: You can simply use <field name="state">open</field> . No need to explicitly call en eval.

File: addons/survey/models/survey_survey.py [hunk 3]
```diff
@@ -21,12 +21,13 @@ class Survey(models.Model):
     _rec_name = 'title'
     _inherit = ['mail.thread', 'mail.activity.mixin']
 
-    def _get_default_stage_id(self):
-        return self.env['survey.stage'].search([], limit=1).id
-
     def _get_default_access_token(self):
         return str(uuid.uuid4())
 
+    def _read_group_states(self, values, domain, order):
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_survey.py]: Guidelines note: just put it after CRUD overrides .

File: addons/survey/models/survey_survey.py [hunk 4]
```diff
@@ -37,9 +38,14 @@ def _get_default_access_token(self):
     question_and_page_ids = fields.One2many('survey.question', 'survey_id', string='Sections and Questions', copy=True)
     page_ids = fields.One2many('survey.question', string='Pages', compute="_compute_page_and_question_ids")
     question_ids = fields.One2many('survey.question', string='Questions', compute="_compute_page_and_question_ids")
-    stage_id = fields.Many2one('survey.stage', string="Stage", default=lambda self: self._get_default_stage_id(),
-                               ondelete="restrict", copy=False, group_expand='_read_group_stage_ids')
-    is_closed = fields.Boolean("Is closed", related='stage_id.closed', readonly=True)
+    state = fields.Selection(
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_survey.py]: It should be required to avoid having surveys in undefined states.

Why putting it readonly ?

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by Xavier-Do** [comment]
robodoo retry

**Pull request closed without merging**